agent/lib/kontena/websocket_client.rb

```diff
@@ -171,6 +172,7 @@ def on_close(event)
         handle_invalid_version
       end
       info "connection closed with code #{event.code}"
+      notify_actors('websocket:close', nil)
```

Comment:
The `websocket:disconnect` happens immediately when the agent decides that it wants to `close` the websocket. The `websocket:close` from `on :close` might come 30s later, and any messages that are sent between those two are presumably just going to get lost.

OTOH if the server shuts down or closes the connection, then there is never going to be a `websocket:disconnect`, but now there is going to be a `websocket:close`.

Are the notification names are confusing enough? There's also `websocket:connect` vs `websocket:connected`, which are very different and come from different places.. :)